src/Templates/NServiceBusWindowsService/EndpointConfigurationExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Diagnostics;
+using NServiceBus;
+
+namespace NServiceBusWindowsService
+{
+    // TODO: move this into a shared project for use in all endpoints
+    internal static class EndpointConfigurationExtensions
+    {
+        public static void ApplySettingsForAllEndpoints(
+            this EndpointConfiguration endpointConfiguration,
+            Action<TransportExtensions<LearningTransport>> applyEndpointSpecificSettings)
+        {
+            // TODO: ensure the most appropriate serializer is chosen
+            // https://docs.particular.net/nservicebus/serialization/
+            endpointConfiguration.UseSerialization<NewtonsoftSerializer>();
+
+            // TODO: remove this condition after choosing a transport, persistence and deployment method suitable for production
+            if (Environment.UserInteractive && Debugger.IsAttached)
+            {
+                // TODO: choose a durable transport for production
+                // https://docs.particular.net/transports/
+                var transportExtensions = endpointConfiguration.UseTransport<LearningTransport>();
+                applyEndpointSpecificSettings?.Invoke(transportExtensions);
```

Comment:
Why is this being invoked here specifically? Why not after everything else in here has been called?

Reply:
Good question, and I have no good answer. I'll move it to the end.